### PR TITLE
Update swiftlint.yml and fix code

### PR DESCRIPTION
**Pull request by tikipatel**
This PR updates the `.swiftlint.yml` and warnings produced by the updated rules.

### CODE REVIEW

File: Aspire Budgeting/AppDelegate.swift [hunk 1]
```diff
@@ -7,29 +7,40 @@ import UIKit
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
```
Comment (mohitathwani) [inline Aspire Budgeting/AppDelegate.swift:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/02f411f4e99e9c92eda83f52b2dcb91b2ca8561d/Aspire Budgeting/AppDelegate.swift#L9"></span>Consider using final for this class or use a struct (final_class)</td>
    </tr>
  </tbody>
</table>

File: Aspire BudgetingTests/GoogleSDKCredentialsTests.swift [hunk 2]
```diff
@@ -10,66 +10,75 @@
 import XCTest
 
 class GoogleSDKCredentialsTests: XCTestCase {
```
Comment (mohitathwani) [inline Aspire BudgetingTests/GoogleSDKCredentialsTests.swift:12]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/02f411f4e99e9c92eda83f52b2dcb91b2ca8561d/Aspire BudgetingTests/GoogleSDKCredentialsTests.swift#L12"></span>Consider using final for this class or use a struct (final_class)</td>
    </tr>
  </tbody>
</table>

**Comment by mohitathwani** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Aspire Budgeting/Facilities/GoogleDriveManager.swift [hunk 3]
```diff
@@ -13,101 +13,126 @@ import GTMSessionFetcher
...
+
+    let backupFileList = fileList
+    fileList.removeAll()
+
+    driveService.authorizer = authorizer
+    driveService.shouldFetchNextPages = true
+
+    googleFilesListQuery.fields = GoogleDriveManager.queryFields
+    googleFilesListQuery.q = "mimeType='\(GoogleDriveManager.spreadsheetMIME)'"
+    // TODO: teeks -- fix line_length
```
[the remark this answers is not in the document]
Comment (mohitathwani) [reply Aspire Budgeting/Facilities/GoogleDriveManager.swift:108]: ahh is it on the line above with the mimeType string?

**Comment by mohitathwani** [comment]
Compiles successfully. Tests fail, but we know that already. All green from my side.

File: Aspire Budgeting/Facilities/StateManager.swift [hunk 4]
```diff
@@ -18,97 +18,139 @@ final class StateManager: ObservableObject {
...
+      )
       self.transition(to: .loggedOut)
-    })
+    }
   }
-  
+
   func transition(to nextState: State) {
-    if canTransition(to: nextState) {
+    if self.canTransition(to: nextState) {
```
[the remark this answers is not in the document]
Comment (tikipatel) [reply Aspire Budgeting/Facilities/StateManager.swift]: Correct. This was an issue with the formatter. There are several other examples of this kind of usage which can be updated as we refactor.

File: Aspire Budgeting/ContentView.swift [hunk 5]
```diff
@@ -14,41 +14,46 @@ struct ContentView: View {
...
     stateManager.currentState == StateManager.State.hasDefaultSheet
   }
-  
+
   var body: some View {
     VStack {
       if isLoggedOut {
-        SignInView().animation(Animation.spring().speed(1.0)).transition(.move(edge: .trailing))
+        SignInView()
+          .animation(Animation.spring().speed(1.0)).transition(.move(edge: .trailing))
```
Comment (tikipatel) [inline Aspire Budgeting/ContentView.swift:38]: @mohitathwani There are also a few examples of this that would need updating -- each modifier should be in its own line.

File: Aspire Budgeting/Facilities/GoogleDriveManager.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (tikipatel) [reply Aspire Budgeting/Facilities/GoogleDriveManager.swift:108]: It's the line below `ticket = .. ` fixing now

**Pull request merged**